File: Chapter4/wcreducer.py
#!/usr/bin/env python
import sys

count = 0
current = None

for line in sys.stdin.readlines():
	word, count2 = line.split("\t")
	if word == current:
		count = int(count) + int(count2)
	else:
		if current != None:
			sys.stdout.write(current.strip() + "\t" + str(count)+"\n")
		current = word
		count = int(count2)
sys.stdout.write(current.strip() + "\t" + str(count)+"\n")

File: Chapter4/wcreducer.rb
#!/usr/bin/env ruby

current = nil
count = 0

while line = gets
	word, count2 = line.split("\t")
	if word == current
		count = count2.to_i+1
	else
		puts current +"\t" + count.to_s if current
		current = word
		count = 1
	end
end
puts current + "\t" + count.to_s

